### PR TITLE
Add support for listing target groups with the space and org tag

### PR DESCRIPTION
**Pull request by fishnix**
Allow getting the list of target groups with the `spinup:spaceid` and `spinup:org` tag for a given space.  This allows for getting a list of available load balancers when provisioning a container service.  The ARN can then be included in the Service definition.

### CODE REVIEW

File: api/handlers_lbs.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	"github.com/YaleSpinup/ecs-api/apierror"
+	"github.com/YaleSpinup/ecs-api/resourcegroupstaggingapi"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+func (s *server) LoadBalancerListHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (damnski) [inline api/handlers_lbs.go:16]: :eyes:  comment comment

**Comment by damnski** [comment]
lgtm

File: resourcegroupstaggingapi/resourcegroupstaggingapi.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials: credentials.NewStaticCredentials(account.Akid, account.Secret, ""),
+		Region:      aws.String(account.Region),
+	}))
+	s.Service = resourcegroupstaggingapi.New(sess)
+	return s
+}
+
+func (r *ResourceGroupsTaggingAPI) GetResourcesWithTags(ctx context.Context, types []string, filters []*TagFilter) ([]string, error) {
+	if len(filters) == 0 {
```
Comment (tenyo) [inline resourcegroupstaggingapi/resourcegroupstaggingapi.go:41]: Should also check that `types` isn't empty

File: resourcegroupstaggingapi/resourcegroupstaggingapi.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	}
+
+	log.Infof("getting resources with type '%s' that match tags", strings.Join(types, ", "))
+
+	tagFilters := make([]*resourcegroupstaggingapi.TagFilter, 0, len(filters))
+	for _, f := range filters {
+		log.Debugf("tagfilter: %s:%+v", f.Key, f.Value)
+		tagFilters = append(tagFilters, &resourcegroupstaggingapi.TagFilter{
+			Key:    aws.String(f.Key),
+			Values: aws.StringSlice(f.Value),
```
Comment (tenyo) [inline resourcegroupstaggingapi/resourcegroupstaggingapi.go:52]: Cool, I didn't know about `StringSlice`

File: resourcegroupstaggingapi/resourcegroupstaggingapi_test.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+		t.Errorf("expected nil error, got %s", err)
+	}
+
+	expected := []string{"arn:aws:elasticloadbalancing:us-east-1:1234567890:targetgroup/testtg123/0987654321"}
+	if !reflect.DeepEqual(expected, out) {
+		t.Errorf("expected %+v, got %+v", expected, out)
+	}
+
+	out, err = r.GetResourcesWithTags(context.TODO(), []string{}, filters)
+	if err != nil {
```
Comment (tenyo) [inline resourcegroupstaggingapi/resourcegroupstaggingapi_test.go:179]: oh, I see empty types might actually be a valid option (that returns all types), should've looked at the test first 😄

**Comment by fishnix** [comment]
mostly tests and error handing @tenyo :)

**Pull request merged**